documentation/synthetic_tokens/security/security.md

```diff
@@ -0,0 +1,107 @@
+Whichever sponsors and token holders redeem first will receive collateral. 
+Whoever comes last will receive less or nothing depending on the amount of collateral left in the contract.
+* Solution: Although we expect the economic incentives to work without a haircut mechanism, if a position does reach < 100% collateralization, UMA will liquidate these positions uneconomically.
+
+## Open issues affecting capital efficiency: 
+### New token creation
+* Scenario: Token sponsors who wish to create new tokens are required to collateralize the new tokens at levels that meet or exceed the global collateralization ratio (GCR). 
+However, this makes the role of token sponsor capital inefficient for 2 reasons: 
+1. If a token sponsor has an outstanding synthetic token position collateralized well above the GCR, such that the creation of new tokens could be supported by the current amount of excess collateral in their position, they cannot use this collateral to create new synthetic tokens. 
+1. If a token sponsor wishes to create new tokens collateralized to a level below the GCR, they must first collateralize their position above the GCR and go through a “slow” withdrawal to withdraw their excess collateral.
+
+### Whale griefing attack
+* Scenario: A griefing sponsor can add a proportionally large amount of collateral relative to the rest of the contract while creating a very small debt position. 
+This drastically imbalances the contract GCR upwards. 
+For example, a malicious sponsor can add 100x the current contract collateral while only creating a 1wei position in debt. 
+This results in the contract GCR drastically increasing (~100x), making all later sponsors required to be massively over collateralized. 
+At best, this slows down future token sponsorship (must repeatedly create and wait on withdraw), and at worst, this can lock all future sponsorship, constricting token supply. 
+This could make it difficult or impossible for liquidators to liquidate positions and simultaneously drive up the token price.
+One worrisome attack vector is if a sponsor were to use this to constrict token supply, and then issue a withdraw request to bring their collateralization ratio to 0%. 
+If liquidators cannot get tokens to liquidate the sponsor’s position during the liveness period, the sponsor could walk away with a profit. 
+There are a number of assumptions that have to be in place for this to work: 
+1. Few or no tokens were minted before this sponsor increased the GCR
+2. The sponsor is better capitalized than any liquidator
+3. A liquidator does not have enough tokens or collateral to issue a partial liquidation large enough to wipe out the profitability of the sponsor. 
+If the sponsor is 100x collateralized, a partial liquidation for 1/100ths of the sponsor’s tokens would essentially wipe out all of the profit.
+Any larger liquidations, and they would lose money after the remaining withdrawal went through. 
```

Comment:
What's the solution?

Reply:
As the OZ doc indicates, the solution is to assume that the assumptions listed above are not met. We can clarify this with a "Conclusion" indicating that so a liquid, well-functioning market for synthetic tokens should not meet any of the assumptions. 

From the OZ doc: 

Solution: assume that legitimate token contracts are liquid enough to make this attack impossibly expensive. More analysis needs to be done here around just how liquid a market needs to be to be safe.